tests/TestDocs.hs

```diff
@@ -106,6 +106,8 @@ data DocsAssertion
   | ShouldHaveLink P.ModuleName Text Text Docs.Namespace Docs.LinkLocation
   -- | Assert that a given declaration comes before another in the output
   | ShouldComeBefore P.ModuleName Text Text
+  -- | Assert that a given declaration has a kind signature
+  | ShouldHaveKindSignature P.ModuleName Text
```

Comment:
Can you please add another argument for the kind signature? We should test not just that the kind signature is present, but also that it is what we expect it to be.

Reply:
Yup. That was the direction I was headed, but I ran out of time.

Also, shouldn't we also be testing that the keyword (i.e. `data`, etc.) is the one we expect?

Reply:
Sounds good! Up to you whether we test the keyword as well. That seems like it would be relatively low impact if we got it wrong, so I’m not sure whether the overhead of testing it would be worth it.

Reply:
I think it's still important. I'd hate to see docs like this:
```
newtype Foo :: forall k. k -> Type
data Foo a = Foo
```
However, I felt it would be easier on the developer to just type in the expected kind signature rather than try to construct the proper one via Docs types.